packages/lib-core/src/runtime/PluginLoader.ts

```diff
@@ -148,54 +153,54 @@ export class PluginLoader {
   /** Plugins processed by this loader. */
   private readonly plugins = new Map<string, PluginLoadData>();
 
-  /** Subscribed event listeners. */
-  private readonly listeners = new Set<PluginLoadListener>();
+  private readonly loadListeners = new Set<VoidFunction>();
 
   constructor(options: PluginLoaderOptions = {}) {
     this.options = {
       canLoadPlugin: options.canLoadPlugin ?? (() => true),
       canReloadScript: options.canReloadScript ?? (() => true),
+      entryCallbackSettings: options.entryCallbackSettings ?? {},
       fetchImpl: options.fetchImpl ?? basicFetch,
       fixedPluginDependencyResolutions: options.fixedPluginDependencyResolutions ?? {},
       sharedScope: options.sharedScope ?? {},
       postProcessManifest: options.postProcessManifest ?? _.identity,
-      entryCallbackName: options.entryCallbackName ?? DEFAULT_REMOTE_ENTRY_CALLBACK,
       getPluginEntryModule: options.getPluginEntryModule ?? _.noop,
     };
+
+    if (this.options.entryCallbackSettings.autoRegisterCallback ?? true) {
```

Comment:
Personally I'd prefer to perform this kind of check outside the function, since
```ts
if (doNothingCondition) {
  return;
}
```
adds complexity without much gain.

I agree that the `if` statement expression in
```ts
if (this.options.entryCallbackSettings.autoRegisterCallback ?? true) {
  this.registerPluginEntryCallback();
}
```
looks a bit weird. I'll try to improve that :smiley: